Makefile.am

```diff
@@ -45,6 +45,10 @@ src_libfabric_la_SOURCES = \
 	src/log.c \
 	$(common_srcs)
 
+if MACOS
+src_libfabric_la_SOURCES += src/osx/osd.c
+endif
+
```

Comment:
The new headers should be added to "src_libfabric_la_SOURCES" as well. Try "make dist" then build from resultant tarball to check.